cpp/src/graph_utils.cuh

```diff
@@ -325,6 +325,15 @@ namespace cugraph
 			if (degree[i] == 0)
 				bookmark[i] = 1.0;
 	}
+
+        template<typename IndexType, typename ValueType>
+        __global__ void __launch_bounds__(CUDA_MAX_KERNEL_THREADS)
+        degree_offsets(const IndexType n, const IndexType e, const IndexType *ind, IndexType *degree) {
+                for (int i = threadIdx.x + blockIdx.x * blockDim.x; i < n; i += gridDim.x * blockDim.x)
+                        atomicAdd(&degree[i], ind[i+1]-ind[i]);
+        }
```

Comment:
I think we do not need an atomic here. Each thread `i` will write its own output only once at the `i`th location in the degree array

Reply:
removed atomic add